src/scripts/main.js

```diff
@@ -1,11 +1,20 @@
 'use strict';
 
 function waitFor(element, eventName) {
-// write your code here
+  return new Promise(resolve => {
+    element.addEventListener(eventName, (e) => {
+      resolve(`It was ${eventName} on the element: ${
+        element.nodeName}, id: ${element.id}.`);
+    });
+  });
 }
 
 const printMessage = (message) => {
-// write your code here
+  document.body.insertAdjacentHTML('beforeend', `
```

Comment:
It's better to create a new 'div' element, set its class and innerHTML, then append it to 'document.body' instead of using 'insertAdjacentHTML'. This avoids potential security risks with injecting HTML. 🛡️